lib/livebook/hubs.ex

```diff
@@ -293,4 +295,35 @@ defmodule Livebook.Hubs do
   def set_offline_hub(hub) do
     :persistent_term.put(@offline_hub_key, hub)
   end
+
+  @doc """
+  Connects the given hub, without connecting with WebSocket.
+  """
+  @spec start_offline_hub(Provider.t(), String.t() | nil) :: :ok | :error
+  def start_offline_hub(hub, encrypted_secrets) do
+    {client, ^hub} = Provider.connection_spec(hub)
+
+    {:ok, pid} =
+      DynamicSupervisor.start_child(@supervisor, %{
+        id: hub.id,
+        start: {client, :start_link, [hub, _offline? = true]}
```

Comment:
Instead of passing offline as a flag, what if we pass the decrypted secrets as second argument, so we have them as the initial server state?

Reply:
If so, we will need to check if the secrets list is not empty to block the WebSocket connection, right?

Reply:
Beucase, if the WebSocket tries to connect and receives an error about invalid credentials, the Hub will be deleted